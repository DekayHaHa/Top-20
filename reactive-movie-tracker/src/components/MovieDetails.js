import React, { Component } from 'react';
import '../styles/MovieDetail.scss'

export const MovieDetails = class extends Component {
	render() {
		const {overview, score, image, title } = this.props
		return (
			<div className="movie-details">
				<img className="img-details" alt='movie poster' src={image} />
				<div className='details-container'>
				<h1 className="title-details">{title}</h1>
				<p className="score-details">Score: {score}</p>
				<p className="overview-details">{overview}</p>
				</div>
			</div>
		);
	}
}