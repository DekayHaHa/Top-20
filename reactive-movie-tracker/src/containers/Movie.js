import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

const Movie = class extends Component {

  addToFavorites = async () => {
    const url = "http://localhost:3000/api/users/favorites/new";
    const movieInfo = {
      movie_id: this.props.id,
      user_id: this.props.activeUser.id,
      title: this.props.title,
      poster_path: this.props.image,
      release_date: this.props.releaseDate,
      vote_average: this.props.score,
      overview: this.props.overview
    };
    const response = await fetch(url, {
      method: "POST",
      body: JSON.stringify(movieInfo),
      headers: {
        "Content-Type": "application/json"
      }
    });
    const data = await response.json();
    
  };

  deleteFromFavorites = async () => {
    const user_id = this.props.activeUser.id;
    const movie_id = this.props.id;
    const url = `http://localhost:3000/api/users/${user_id}/favorites/${movie_id}`;
    const movieInfo = {
      movie_id: this.props.id,
      user_id: this.props.activeUser.id
    };
    const response = await fetch(url, {
      method: "DELETE",
      body: JSON.stringify(movieInfo),
      headers: {
        "Content-Type": "application/json"
      }
    });
    const data = await response.json();
    
  };


  render() {
		const { id, title, image } = this.props;

    return (
      <div>
        <Link to={`/movie/${id}`}>
          <h1>{title}</h1>
          <img alt="movie poster" src={image} />
        </Link>
        <button onClick={this.addToFavorites}>Favorite</button>
        <button onClick={this.deleteFromFavorites}>Unfavorite</button>
      </div>
    );
  }
};

export const mapStateToProps = state => ({
  movies: state.movies,
  activeUser: state.activeUser
});

export default connect(
  mapStateToProps,
  null
)(Movie);
