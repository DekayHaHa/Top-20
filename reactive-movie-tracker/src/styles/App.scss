

html {
  background-image: url('../styles/movie-theatre.jpeg');
  background-repeat: no-repeat;
  background-position: center;
  margin: none;
  padding: none;
  font-family: 'Cabin', sans-serif;
}