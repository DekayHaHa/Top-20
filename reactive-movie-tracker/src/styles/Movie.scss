.movie {
	width: 33%;
	border: 1px solid black;
	margin: 20px 0px;
	position: relative;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	padding: 10px;
}

.movie-title {
	text-align: center;
	margin: 0px;
}

img {
	width: 100%;
}

.favorite-btn {
	// width: 65px;
	// height: 65px;
	position: absolute;
	top: 14px;
	left: 10px;
	background: none;
	border: none;
	// background-image: url('./iconmonstr-star-6.svg')

	&:hover {
    background: lightseagreen;
  }
}