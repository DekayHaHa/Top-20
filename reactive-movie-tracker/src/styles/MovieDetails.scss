.movie-details {
	color: white;
	display: flex;
	text-shadow: 2px 2px 2px black;
}

.img-details {
	width: 30%;
	margin: 50px 50px auto 50px;
}

.details-container {
	margin: 50px 50px auto 50px;
}

.title-details {
	margin: 0px;
}

.score-details, .overview-details {
	font-size: 20px;
}