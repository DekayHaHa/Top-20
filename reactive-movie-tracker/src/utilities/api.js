import { cleanFavorites } from "./cleaner";

export const fetchData = url => {
  return fetch(url)
    .then(response => response.json())
    .catch(error => error.message);
};

export const retrieveAllFavorites = async url => {
  const response = await fetch(url);
  if (!response.ok) {
    throw Error(response.statusText);
  } else {
    const data = await response.json();
    return await cleanFavorites(data.data);
  }
};


