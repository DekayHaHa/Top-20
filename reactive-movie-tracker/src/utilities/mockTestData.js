export const mockReduxMovies = [
	{	title: 'Kung Fu',
	image: 'www.reddit.com',
	overview: 'Big Blue House',
	score: 7.8,
	id: 166428,
	isFavorite: false,
	releaseDate: "2019-01-03",},
	{	title: 'Kung Fu',
	image: 'www.reddit.com',
	overview: 'Big Blue House',
	score: 7.8,
	id: 166428,
	isFavorite: false,
	releaseDate: "2015-01-03",},
	{	title: 'Spider Man',
	image: 'www.reddit.com',
	overview: 'Big Blue House',
	score: 9,
	id: 123455,
	isFavorite: false,
	releaseDate: "2017-01-03",},
]

export const mockReduxUser = {
	name: 'Jeff',
	id: 2,
}

export const mockReduxFavs = [
	{title: 'Spider Man',
	image: 'www.reddit.com',
	overview: 'Big Blue House',
	score: 9,
	id: 123455,
	isFavorite: true,
	releaseDate: "2017-01-03",}
]