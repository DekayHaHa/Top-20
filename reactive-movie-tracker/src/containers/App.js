import React, { Component } from "react";
import MovieHolder from "./MovieHolder";
import { fetchData } from "../utilities/api";
import { APIkey } from "../utilities/key.js";
import { connect } from "react-redux";
import { addMovies } from "../actions";
import Header from "./Header";
import { Route, Switch } from "react-router-dom";
import Login from "./Login";
import { cleanMovies } from "../utilities/cleaner";
import { MovieDetails } from "../components/MovieDetails";
import { PageNotFound } from "../components/PageNotFound";
import PropTypes from "prop-types";
import "../styles/App.scss";

export class App extends Component {
  componentDidMount() {
    this.getMovieData();
  }
  getMovieData = async () => {
    const url = `https://api.themoviedb.org/3/movie/popular?api_key=${APIkey}&language=en-US&page=1`;
    try {
      //isloading
      const response = await fetchData(url);
      const finalData = cleanMovies(response.results);
      this.props.addMovies(finalData);
    } catch(error) {
      //finish out
    }
  };
  findMovieToRender = id => {
    const MovieToRender = this.props.movies.find(movie => movie.id === id);
    const movie = <MovieDetails {...MovieToRender} />;
    return MovieToRender ? movie : false;
  };
  render() {
    return (
      <div className="App">
        <Route path="/" component={Header} />
        <Switch>
          <Route exact path="/" component={MovieHolder} />
          <Route exact path="/login" component={Login} />
          <Route
            exact
            path="/movie/:id"
            render={({ match }) => {
              const movie = this.findMovieToRender(parseInt(match.params.id));
              return movie || <PageNotFound />;
            }}
          />
          <Route component={PageNotFound} />
        </Switch>
      </div>
    );
  }
}

App.propTypes = {
  movies: PropTypes.array,
  activeUser: PropTypes.object
};

export const mapStateToProps = state => ({
  movies: state.movies,
  activeUser: state.activeUser
});

export const mapDispatchtoProps = dispatch => ({
  addMovies: movies => dispatch(addMovies(movies))
});

export default connect(
  mapStateToProps,
  mapDispatchtoProps
)(App);
