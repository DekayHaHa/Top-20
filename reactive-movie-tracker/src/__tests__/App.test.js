import React from 'react';
import { shallow } from "enzyme";
import {App, mapStateToProps, mapDispatchtoProps} from '../containers/App'
import {fetchData} from '../utilities/api'
import { mockReduxFavs, mockReduxMovies, mockReduxUser } from '../utilities/mockTestData'
import { MovieDetails } from '../components/MovieDetails'
import Movie from '../containers/Movie';

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App 
      movies={mockReduxMovies}
      activeUser={mockReduxUser}
      favorites={mockReduxFavs}
    />);
  });


  it("should match the snapshot with all data passed in correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip("should add movie data to state", () => {
    //setup

    //execution 
    wrapper.instance().getMovieData()
    //expectation
  });

  it("should render movie based on id", () => {
    //setup
    const movie = {
      title: 'Spider Man',
      image: 'www.reddit.com',
      overview: 'Big Blue House',
      score: 9,
      id: 123455,
      isFavorite: false,
      releaseDate: "2017-01-03",
    }
    const expected = <MovieDetails {...movie}/>
    //execution 
    const results = wrapper.instance().findMovieToRender(123455)
    //expectation
    expect(results).toEqual(expected)
  });

});