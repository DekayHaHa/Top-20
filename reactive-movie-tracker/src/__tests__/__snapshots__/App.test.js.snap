// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div
  className="App"
>
  <Route
    component={[Function]}
    path="/"
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      exact={true}
      path="/movie/:id"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
